enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1220 @@
+
+### Separate control-plane and worker-node updates into separate phases, but implement a simpler MaintenanceSchedule strategy
+We could implement change control without `disabledUntil`, `pausedUntil`, `exclude`, and perhaps more. However,
+it is risky to impose a single opinionated workflow onto the wide variety of consumers of the platform. The workflows
+described in this enhancement are not intended to be exotic or contrived but situations in which flexibility
+in our configuration can achieve real world, reasonable goals. 
+
+`disabledUntil` is designed to support our Service Delivery team who, on occasion, will need
+to be able to bypass configured change controls. The feature is easy to use, does not require 
+deleting or restoring customer configuration (which may be error-prone), and can be safely 
+"forgotten" after being set to a date in the future.
+
+`pausedUntil`, among other interesting possibilities, offers a cluster lifecycle administrator the ability
+to stop a problematic update from unfolding further. You may have watched a 100 node
+cluster roll out a bad configuration change without knowing exactly how to stop the damage
+without causing further disruption. This is not a moment when you want to be figuring out how to format
+a date string, calculating timezones, or copying around cluster configuration so that you can restore
+it after you stop the bleeding.
+
+### Implement change control, but do not implement the Manual and/or Assisted strategy for MachineConfigPool
+Major enterprise users of our software do not update on a predictable, recurring window of time. Updates
+require laborious certification processes and testing. Maintenance schedules will not serve these customers
+well. However, these customers may still benefit significantly from the change management concept --
+unexpected / disruptive worker node drains and reboots have bitten even experienced OpenShift operators
+(e.g. a new MachineConfig being contributed via gitops).
+
+These strategies inform decision-making through metrics and provide facilities for fine-grained control 
+over exactly when material change is rolled out to a cluster. 
+
+The Assisted strategy is also specifically designed to provide a foundation for 
+the forthcoming `oc adm update worker-nodes` verbs. After separating the control-plane and
+worker-node update phases, these verbs are intended to provide cluster lifecycle administrators the 
+ability to easily start, pause, cancel, and even rollback worker-node changes.
+
+Making accommodations for these strategies should be a subset of the overall implementation
+of the MaintenanceSchedule strategy and they will enable a foundation for a range of 
+different persons not served by MaintenanceSchedule.
+
+### Use CRON instead of RRULE
+The CRON specification is typically used to describe when something should start and 
+does not imply when things should end. CRON also cannot, in a standard way,

```

Comment:
I thought we were excluding the ability to specify an end date in RRULE anyway?

Reply:
I should probably add to the deciding factors "it is not as expressive" (e.g. cron cannot express "last Sunday of the month"). But you bring up a good point that deserves clarification. Here, "end" refers to the closing of a permissive window.  `*/5 * * * *` means invoke something every 5 minutes (immaterial of when the task finishes). We'd need to contort this to mean "unpaused for the duration of every 5th minute". 

The RRULE restriction you mention refers to  "end" as in "the schedule no longer applies after a (date X | X repititions)". For example, in google calendar, you can schedule a reoccurring meeting that only reoccurs for 6 weeks. The schedule ends / is irrelevant, after those 6 weeks.
I did not see a use case for allowing such an "end" to be configured since it seemed at odds to keeping a cluster secure.

Reply:
How are the permissive windows reflected in RRULE, it has a similar problem to cron no? It's about when to trigger but doesn't specify a duration for the window?

Reply:
RRULE selects days. That provides a intuitive (default) 24h bound for start and stop time. The EP allows for further refinement of this with intra-day `startTime` and `duration`.

Reply:
You could pair a `duration`-like concept with cron, but you are back to contortions: `* 20-23 * * *` might be a natural first attempt at saying "changes are ok between 20:00 and 23:59" each day; what is duration supposed to be?
If I was trying to make it work, I'd say duration defaults to 0 and only times selected by the cron apply. If duration is non-zero, it implies the changes can continue on after the cron stops selecting the time. So if you set duration=1h with `* 20-23 * * *`, that actually implies "20:00 to 1:59" the next day.
cron=`* 20-23 * * *` with duration=0 could then also be expressed as cron=`0 20 * * *`, duration=4h. If you messed up with cron=`* 20 * * *`, duration=4h, your total duration would be 5h. 

Playing around with potential misfires on a simple scenario like "permit updates all day Sunday".
- duration=24h, cron=`* * * * Sun` -- enjoy your Monday updates!
- duration=24h, cron=`0 * * * Sun` -- enjoy your Monday updates!
- duration=24h, cron=`* 0-23 * * Sun` -- enjoy your Monday updates!
- duration=0, cron=`0 0 * * Sun` -- hope your updates can complete in 1 minute!
- duration=0, cron=`* 0 * * Sun` -- nope, that's an hour on Sunday
- duration=0, cron=`0 0-23 * * Sun` -- nope, that's 24 non-contiguous minutes on Sunday
- duration=24h, cron=`0 0 * * Sun` -- success!
- duration=0, cron=`* * * * Sun` -- success!

I find it very difficult to reason about. 

If I try to play the same game with RRULE malfeasance in this EP. An off the shelf tool tells the user `FREQ=WEEKLY;INTERVAL=1;BYDAY=SU`
- No other options set? -- success!
- duration=24h -- success!
- startTime=00:00 -- success!
- startTime=00:00, duration=24h -- success!

Reply:
Well that's a fantastic argument for not using cron 😅 

If the API we are exposing only allows selecting particular days I think that simplifies things significantly for us. Do end users care about specific sub day timing? Eg would they want it to be in their normal waking hours? Is that something we are willing to support?

Reply:
Right, so we are saying that users will specify an RRULE, a start time and a duration, that makes me lean even more towards breaking the API for RRULE down into something Kube native tbh. Will come up with something